### DISCUSSION TITLE
Apple TN3147 - altool Deprecation notification.

### DISCUSSION THREAD

**Discussion by eleanor-naim** [fastlane/fastlane#21347] [Q&A] [answered]
I received an email from Apple the other day: 

_We’re reaching out because you recently used the altool command-line utility to notarize your macOS software with Apple. As announced last year at [WWDC22](https://developer.apple.com/videos/play/wwdc2022/10109), if you’re still using altool with the Apple notary service, you should transition to the notarytool command-line utility as soon as possible. Notarizing software with altool was deprecated in Xcode 13, and the Apple notary service will no longer accept uploads from altool as of November 1, 2023. Existing notarized software will continue to function properly. 
For information on notarizing your apps, read [TechNote TN3147: Migrating to the latest notarization tool](https://developer.apple.com/documentation/technotes/tn3147-migrating-to-the-latest-notarization-tool?changes=_3_3)._

We use _upload_to_testflight_ (Fastlane script) to do our uploading. I cannot see anywhere in our own scripts which mention _altool_ so I assume this is used within this script to do our notarisation, since moving from iTMSTransporter last year. We use an API_KEY when connecting to Apple and everything is working great. 

Is there any plan on fixing _upload_to_testflight_ to use _notarytool_ or it this not actually an issue for us going forward?

Any guidance is greatly appreciated.

**Comment by eleanor-naim** [comment]
Can anyone advise on this?

**Comment by eleanor-naim** [comment]
We are now < 1 month before _upload_to_testflight_ might be broken. Can I assume the lack of any guidance means I should look for an alternative solution to address this problem?

**Comment by TheMetalCode** [comment, marked as answer]
This tidbit from the [Apple tech note](https://developer.apple.com/documentation/technotes/tn3147-migrating-to-the-latest-notarization-tool?changes=_3_3) seems like a potentially important distinction:

```
Important

Apple has deprecated altool for the purposes of notarization and [announced that it will stop working for notarization on 2023-11-01](https://developer.apple.com/news/?id=y5mjxqmn). However, altool is still a good way to perform other tasks, like submitting an app to the App Store.
```

It does not sound like `altool` is being _outright_ deprecated, but deprecated only for notarization. I suspect that the `pilot` and `deliver` actions can still use `altool` for app uploads, appstore submission, etc.

The [`notarize` action](https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/notarize.rb#L41) seems to already allow for the usage of `notarytool` instead of `altool`, and the [default value](https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/notarize.rb#L268) of the `use_notarytool` param is `true` if you're using Xcode 13.0 or higher.
